crates/sovereign-sdk/full-node/db/sov-db/src/ledger_db/mod.rs

```diff
@@ -564,6 +564,23 @@ impl LedgerDB {
         Ok(())
     }
 
+    /// Set the last scanned slot by the prover
+    /// Called by the prover.
+    #[instrument(level = "trace", skip_all, err, ret)]
+    pub fn set_l2_witness<Witness: Serialize>(
+        &self,
+        l2_height: u64,
+        witness: &Witness,
+    ) -> anyhow::Result<()> {
+        let buf = bincode::serialize(witness)?;
```

Comment:
`SequencerCommitment`did not need a serialization, are we sure we need this? also I thought we said we'd use borsh, why bincode?

Reply:
> SequencerCommitment did not need a serialization, are we sure we need this?

Because `SequencerCommitment` is a concrete type. Inside prover we don't have it, we have only some generic associated type.

Bincode instead of borsh because borsh requires BorshSerialize instead of Serialize. The Serialize trait is implemented for most of types in the project.